controllers/office.js

```diff
@@ -0,0 +1,46 @@
+const OfficeModel = require('../models/office')
+// module for get all data from the office data
+module.exports.get = function get (req, res) {
+  OfficeModel.find({}).exec(function (err, offices) {
+    if (err) return res.status(500).send()
+    res.json({offices})
+  })
+}
+
+// module for get only one data from the office data through the id
+module.exports.get1 = function get (req, res) {
+  OfficeModel.findOne({ _id: req.params.id }, req.body)
+  .exec(function (err, data) {
+    if (err) return res.status(500).send(err)
+    res.json({data})
+  })
+}
+
+// module for create a new data to the office data
+module.exports.post = function post (req, res) {
+  const newOffice = new OfficeModel(req.body)
+  newOffice.save(function (err, list) {
+    if (err) return res.status(500).send(err)
+    res.json({
+      _id: list._id
+    })
+  })
+}
+
+// module for update a data from the office data through the id
+module.exports.put = function put (req, res) {
+  OfficeModel.findOneAndUpdate({ _id: req.params.id }, req.body)
+    .exec(function (err, data) {
+      if (err) return res.status(500).send()
+      res.send()
+    })
+}
+
+// module for delete a data from the office data through the id
+module.exports.delete = function del (req, res) {
+  OfficeModel.findOneAndRemove({ _id: req.params.id }, req.body)
```

Comment:
we don't want to delete the row from the database, we want to make a "soft-delete" just to inactivate the row in the database. for example instead of use findOneAndRemove, use findOneAndUpdate and modify a flag called "isActive", remember to add this field also in all the models